src/sftp/handler_test.go

```diff
@@ -12,12 +11,108 @@ import (
 	"github.com/stretchr/testify/mock"
 	yekazip "github.com/yeka/zip"
 	"io"
-	"log/slog"
 	"os"
 	"path/filepath"
 	"testing"
 )
 
+func Test_NewSFTPHandler_UnableToGetSFTPServerPublicKeyNameSecret_ReturnsError(t *testing.T) {
+	buffer := utils.SetupLogger()
+
+	mockCredentialGetter := new(mocks.MockCredentialGetter)
+	mockCredentialGetter.On("GetSecret", mock.Anything).Return("test123", errors.New("error"))
+
+	sftpHandler, err := NewSftpHandler(mockCredentialGetter)
+
+	assert.Nil(t, sftpHandler)
+	assert.Error(t, err)
+	assert.Contains(t, buffer.String(), "Unable to get SFTP_SERVER_PUBLIC_KEY_NAME")
+}
+
+func Test_NewSFTPHandler_UnableToGetSSHClientHostKeyCallback_ReturnsError(t *testing.T) {
+	buffer := utils.SetupLogger()
+
+	mockCredentialGetter := new(mocks.MockCredentialGetter)
+
+	mockCredentialGetter.On("GetSecret", mock.Anything).Return("test123", nil)
+
+	sftpHandler, err := NewSftpHandler(mockCredentialGetter)
+
+	assert.Nil(t, sftpHandler)
+	assert.Error(t, err)
+
+	assert.Contains(t, buffer.String(), "Unable to get SSH Client Host Key")
+}
+
+func Test_NewSFTPHandler_UnableToGetSFTPUserNameSecret_ReturnsError(t *testing.T) {
+	buffer := utils.SetupLogger()
+
+	mockCredentialGetter := new(mocks.MockCredentialGetter)
+
+	serverKey := "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDg90HXaJnI1KtfJp8MWHxAwC00PvQCZKm4FRRdPGhEMepXIeLdjOtZV6LdePMT3WUmNkd6vaJ4EEmFUtH9lKLidALL9blOJF1iZKXK81JBJsds8axz5cqAau6aclgc9B1z2tAa+JtaSqN7uXvfPsrmsVss4jcOxX+thAhz7U6chN6ahabgIPqHBEjwvPlVNNbSqv0Q0eS4WaEEo/39tiXn5DYpPRC6DjuZ3m5s3VIgHznTv2Ufp3kcLcfEDZFwjm5XRWLNNvM5h3aW1vmr4lgBwuEzPV7CYIdIyDxe9V7YYcGfO+uu/VrDpY1wSmcD3lzHLLTbi5WWOurwiMsWIVRZfa/rmzuoTYknd5iJoiTyIWmR7L0FLfzPlDYJZmAWSdLZrZaUdD8SDIoKMSEV/5/ZzcI0wuoknis+zpyFqT0jfOy7E4GtG8pEQf7JGXaiExNd9TKxbRmaxp3Yv4WgPBThY39Va7EMUC/s0hX2Ah8pIWZG4Lze4x7Z4dElCOHDgnsl3Akc399jnIDfUY4bVn+rfBJntx9mBRaNnV1GqRodbSkHK5dTcZEmRslhuhsQVO2CxrlkPhFEe0XXpA3llO9YIkf4sCZDUbRFKPJiHyDhfrf2/HzkLndODdFaAnICYd51zOI1SgP3aFx60bZ2nPSoLs9DsR1LLIpz4uoiy5hCHw== sschuresko@flexion-mac-J40DPF4YQR"
```

Comment:
If we're using this value a bunch of places, it would help readability to split it out into a constant that can be reused

Reply:
James split it into a constant!